### DISCUSSION TITLE
Imagenet_fake_data_benchmark.py has no split info

### DISCUSSION THREAD

**Discussion by mattiasmar** [google/flax#1773] [Q&A]
What are the prerequisites for running the `Imagenet_fake_data_benchmark.py`?
The `tfds.builder` used in [train.py](https://github.com/google/flax/blob/main/examples/imagenet/train.py)  doesn't generate any split info and thus the method `create_input_iter` in the flax example fails.

Without a disk quota to download all of imagenet, do you have a suggestion on how to run the fake data example?

**Comment by mattiasmar** [comment]
Seems the expected way is to manually download the metadata file [dataset_info.json](https://github.com/tensorflow/datasets/blob/v4.4.0/tensorflow_datasets/testing/metadata/imagenet2012/5.1.0/dataset_info.json) to the folder  `os.path.join(data_dir,imagenet2012/5.1.0)` (where `data_dir` is defined in the beginning of the method `test_fake_data` in `imagenet_fake_data_benchmark.py`, [line 42](https://github.com/google/flax/blob/fb4f2a455d4cae383f11d77707bf0d0f18a45e70/examples/imagenet/imagenet_fake_data_benchmark.py#L42), and `imagenet2012/5.1.0` is defined in the example's [config file](https://github.com/google/flax/blob/fb4f2a455d4cae383f11d77707bf0d0f18a45e70/examples/imagenet/configs/default.py#L40).

@avital @marcvanzee We could automate this or append the example source code with the metadata file (for each of the flax examples). I could submit a PR, but maybe first you could share your thoughts on design / necessity / pitfalls of this issue?

**Reply by avital** [reply]
This script should download the metadata file, does it work for you? https://github.com/google/flax/blob/main/tests/download_dataset_metadata.sh

**Reply by mattiasmar** [reply]
I can see how that script should work, but it doesn't for me. In my organization we use http/https proxies but despite having those set I get a connection error:

This is the output that I get:
```
TFDS metadata does not exist. Downloading...
svn: E170013: Unable to connect to a repository at URL 'https://github.com/tensorflow/datasets/trunk/tensorflow_datasets/testing/metadata'
svn: E000110: Error running context: Connection timed out
```
There are explanations in Stackoverflow on how to use git 2.19 to do a "sparse" checkout. I can try that out tomorrow.

**Reply by mattiasmar** [reply]
This script is a git (v2.19+) alternative that works (for me):
```
#!/bin/bash                                                                                                  
                                                                                                             
set -e                                                                                                       
                                                                                                             
# Download TFDS metadata to flax/.tdfs/metadata directory.                                                   
# This allows the tests to specify the `data_dir` when using tfds.testing.mock_data().                       
cd "$( dirname "$0" )"                                                                                       
                                                                                                             
if [ -d "../.tfds/metadata" ]; then                                                                          
  echo 'TFDS metadata already exists.';                                                                      
else                                                                                                         
  echo 'TFDS metadata does not exist. Downloading...';                                                                                         
  git clone --depth 3  --filter=blob:none   --sparse   https://github.com/tensorflow/datasets/               
  cd datasets                                                                                                
  git sparse-checkout set tensorflow_datasets/testing/metadata                                               
  mkdir ../../.tfds                                                                                          
  mv tensorflow_datasets/testing/metadata/ ../../.tfds/metadata/                                             
  cd ..                                                                                                      
  rm datasets -rf                                                                                            
fi             
```

**Reply by mattiasmar** [reply]
@avital Do you see a value in adding this alternative download method to the project?

**Reply by avital** [reply]
If indeed this works and only clones the metedata file w/o the whole git repo, and removes our dependency on svn, that sounds like an excellent idea to me! How long does the git clone take here?

**Reply by mattiasmar** [reply]
I measured it to 10.3s.

**Reply by avital** [reply]
That's excellent. Then let's use your proposal! Can you please file a pull request (that also removes all references and dependencies to svn in our scripts?) Thanks!

**Reply by mattiasmar** [reply]
I'll give it a shot. Will begin next week.

**Reply by avital** [reply]
Excellent, thanks Mattias!

**Reply by mattiasmar** [reply]
Change done, but I'm not able to run the coverage tests (independent of my own changes).
```
Exception occurred:
  File "/usr/lib/python3/dist-packages/sphinx/jinja2glue.py", line 233, in get_source
    raise TemplateNotFound(template)
```

Do you want to go on a call to quickly sort it out?

**Reply by marcvanzee** [reply]
@mattiasmar I think we totally dropped the ball on this one, apologies! Is this problem still occurring?

**Comment by mattiasmar** [comment]
Don't know. Haven't worked on it since then.
________________________________
